docs/src/aws-construct-lib.rst

```diff
@@ -103,9 +103,274 @@ were part of your app.
+advanced queries. For example, we can look for a child that has a certain CloudFormation resource
+type:
+
+.. code-block:: ts
+
+   const bucketResource =
+      bucket.children.find(c => (c as cdk.Resource).resourceType === 'AWS::S3::Bucket')
+      as s3.cloudformation.BucketResource;
+
+From that point, users are interacting with the strong-typed L1 resources (which
+extend :py:class:`cdk.Resource <@aws-cdk/cdk.Resource>`), so we will look into
+how to extend their surface area to support the various requirements.
+
+Resource Options
+----------------
+
+:py:class:`cdk.Resource <@aws-cdk/cdk.Resource>` has a few facilities for
+setting resource options such as ``Metadata``, ``DependsOn``, etc.
+
+For example, this code:
+
+.. code-block:: ts
+
+   const bucketResource = bucket.findChild('Resource') as s3.cloudformation.BucketResource;
+
+   bucketResource.options.metadata = { MetadataKey: 'MetadataValue' };
+   bucketResource.options.updatePolicy = {
+      autoScalingRollingUpdate: {
+         pauseTime: '390'
+      }
+   };
+
+   bucketResource.addDependency(otherBucket.findChild('Resource') as cdk.Resource);
+
+Will synthesize the following template:
+
+.. code-block:: json
+
+   {
+      "Type": "AWS::S3::Bucket",
+      "DependsOn": [ "Other34654A52" ],
+      "UpdatePolicy": {
+         "AutoScalingRollingUpdate": {
+               "PauseTime": "390"
+         }
+      },
+      "Metadata": {
+         "MetadataKey": "MetadataValue"
+      }
+   }
+
+Overriding Resource Properties
+------------------------------
+
+Each low-level resource in the CDK has a strongly-typed property called
+``propertyOverrides``. It allows users to apply overrides that adhere to the
+CloudFormation schema of the resource, and use code-completion and
+type-checking.
+
+You will normally use this mechanism when a certain feature is available at the
+CloudFormation layer but was not exposed by the AWS Construct.
+
+The following example sets a bucket's analytics configuration:
+
+.. code-block:: ts
+
+   bucketResource.propertyOverrides.analyticsConfigurations = [
+      {
+            id: 'config1',
+            storageClassAnalysis: {
+               dataExport: {
+                  outputSchemaVersion: '1',
+                  destination: {
+                        format: 'html',
+                        bucketArn: otherBucket.bucketArn // use tokens freely
+                  }
+               }
+            }
+      }
+   ];
+
+Raw Overrides
+-------------
+
+In cases the strongly-typed overrides are not sufficient, or, for example, if
+the schema defined in CloudFormation is not up-to-date, the method
+:py:meth:`cdk.Resource.addOverride(path, value) <@aws-cdk/cdk.Resource.addOverride>`
+can be used to define an override that will by applied to the resource
+definition during synthesis.
+
+For example:
+
+.. code-block:: ts
+
+   // define an override at the resource definition root
+   bucketResource.addOverride('Transform', 'Boom');
```

Comment:
We need a better, more real-world, example. What do "Transform" and "Boom" represent?